src/plugins/intel_cpu/tests/functional/single_layer_tests/instances/x64/reduce.cpp

```diff
@@ -1,217 +1,52 @@
-        }
-        auto reductionAxesNode = std::dynamic_pointer_cast<ngraph::Node>(
-                std::make_shared<ngraph::opset3::Constant>(ngraph::element::Type_t::i64, ngraph::Shape(shapeAxes), axes));
-
-        const auto reduce = ngraph::builder::makeReduce(paramOuts[0], reductionAxesNode, keepDims, reductionType);
-
-        selectedType = getPrimitiveType() + "_" +
-                       (inPrc == ElementType::boolean ? "I8" : InferenceEngine::details::convertPrecision(inPrc).name());
-
-        // hybrid layouts
-        if (inFmts.size() != 0 && outFmts.size() == 0) {
-            size_t outShapeSize = inputDynamicShapes[0].size() - axes.size();
-            switch (outShapeSize) {
-                case 0:
-                case 1:
-                    outFmts.push_back(x);
-                    break;
-                case 2:
-                    outFmts.push_back(nc);
-                    break;
-                case 3:
-                    outFmts.push_back(tnc);
-                    break;
-                case 4:
-                    outFmts.push_back(nchw);
-                    break;
-                default:
-                    FAIL() << "Invaid outShapeSize: " << outShapeSize;
-            }
-        }
-
-        function = makeNgraphFunction(netPrecision, params, reduce, "Reduce");
-    }
-
-    void generate_inputs(const std::vector<ngraph::Shape>& targetInputStaticShapes) override {
-        inputs.clear();
-        const auto& funcInputs = function->inputs();
-        for (size_t i = 0; i < funcInputs.size(); ++i) {
-            const auto& funcInput = funcInputs[i];
-            ov::Tensor tensor;
-            if (reductionType == ngraph::helpers::ReductionType::Prod) {
-                tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i], 10, 5);
-                if (netPrecision == ElementType::f32) {
-                    auto *rawBlobDataPtr = static_cast<float *>(tensor.data());
-                    for (size_t i = 0; i < tensor.get_size(); ++i) {
-                        rawBlobDataPtr[i] /= 10.f;
-                    }
-                } else if (netPrecision == ElementType::bf16) {
-                    auto *rawBlobDataPtr = static_cast<ngraph::bfloat16 *>(tensor.data());
-                    for (size_t i = 0; i < tensor.get_size(); ++i) {
-                        rawBlobDataPtr[i] /= 10.f;
-                    }
-                }
-            } else {
-                tensor = ov::test::utils::create_and_fill_tensor(funcInput.get_element_type(), targetInputStaticShapes[i]);
-            }
-
-            inputs.insert({funcInput.get_node_shared_ptr(), tensor});
-        }
-    }
-
-private:
-    ngraph::helpers::ReductionType reductionType;
-    ElementType netPrecision;
+std::vector<std::vector<ov::test::InputShape>> inputShapes_dyn = {
+    {{{{1, 5}, 19, {1, 5}, {1, 10}}, {{2, 19, 2, 2}, {2, 19, 2, 9}}}},
```

Comment:
Just wondering, why we don't support dynamic shapes for reduce in case of ACL?

Reply:
Dynamic shapes are not supported by ConvertReduceMultiAxisBase: https://github.com/openvinotoolkit/openvino/blob/0541a12730b31fe8811811a26dc016bfa4e79316/src/plugins/intel_cpu/src/transformations/cpu_opset/arm/pass/convert_reduce_multi_axis.cpp#L24

Reply:
Isn't it enough to just check the second input (reduce axes)?
I think the transformation should work if only the first input is dynamic

Reply:
Dynamic shapes tests are added to common section (smoke_Reduce_OneAxis_dynamic_CPU and params_MultiAxis_4D_dynamic)